### DISCUSSION TITLE
GCC without build system

### DISCUSSION THREAD

**Discussion by jgromes** [github/securitylab#171] [General] [answered]
Hi,

I'm trying to run CodeQL on of my repositories (https://github.com/jgromes/RadioLib), but I'm seeing the following error message:

`No source code was seen during the build.`

The workflow run in question can be found here: https://github.com/jgromes/RadioLib/runs/1055172671

I suspect this is due to the non-standard build process I'm using - the code is compiled using a command-line utility (arduino-cli), which internally uses avr-gcc. There's no use of build systems like cmake. The output is a GCC-compiled ELF file.

Is there something that's not configured correctly here, or is it required to use e.g. cmake or make to compile prior to CodeQL scan?

Thank you for any help or pointers.

**Comment by igfoo** [comment, marked as answer]
Hi jgromes,

The problem is caused by avr-g++ being executed directly from a statically linked binary. If you do something like
```
for i in ~/.arduino*/packages/arduino/tools/avr-gcc/*/bin/*; do mv "$i" "$i.real"; printf '#!/usr/bin/sh\nexec "'"$i"'.real" ${1+"$@"}\n' > "$i"; chmod +x "$i"; done
```
after `arduino-cli core install arduino:avr` then it should work around the problem.

**Reply by jgromes** [reply]
Thank you for the response - I tried just copy-pasting it (like the noob that I am), but the build step then fails with

```
Error during build: fork/exec /home/runner/.arduino15/packages/arduino/tools/avr-gcc/7.3.0-atmel3.6.1-arduino7/bin/avr-g++: no such file or directory
```

Fortunately it's just the case of specifying interpreter, here's what I'm using:

```bash
for i in ~/.arduino*/packages/arduino/tools/avr-gcc/*/bin/*; do
  mv "$i" "$i.real";
  printf '#!/bin/bash\nexec "'"$i"'.real" ${1+"$@"}\n' > "$i"
  chmod +x "$i";
done
```

Now it works great!
